harmony/harmony.py

```diff
@@ -539,8 +540,47 @@ def download_all(self,
             The filename and path.
         """
         urls = self.result_urls(job_id, show_progress=False) or []
-        file_names = []
+        futures = []
         for url in urls:
             future = self.executor.submit(self._download_file, url, directory, overwrite)
```

Comment:
nit (and not code you touched in this PR, and doesn't need to be fixed, but while we're here)... this whole method could and probably should be:
```python
return [self.download(url, directory, overwrite) for url in self.result_urls(job_id)]
```

Reply:
That's cool. I've occasionally introduced a variable in places like that to assist in debugging but I agree it's nicer not to populate the namespace any more than is necessary.

Reply:
Sure, extracting into a var would be alright, too.  I think the important piece is to avoid repeating the body of `download` inside of `download_all` so that if we change how downloading works we don't have to remember to do it in 2 places.